administrator/components/com_content/models/article.php

```diff
@@ -680,6 +680,39 @@ public function featured($pks, $value = 0)
 		return true;
 	}
 
+	public function shareToken($token)
+	{
+
+		// perform whatever you want on each item checked in the list
+		$jinput = JFactory::getApplication()->input;
+
+		$jform = $jinput->getArray(array('jform' => array('title' => 'string')));
+		$title = $jform['jform']['title'];
+
+		// Get a db connection.
+		$db = $this->getDbo();
+
+		// Create a new query object.
+		$query = $db->getQuery(true);
+
+		// Insert columns.
+		$columns = array('articleId', 'title', 'sharetoken');
+
+		// Insert values.
+		$values = array($db->quote($jinput->get("id")), $db->quote($title), $db->quote($token));
+
+		// Prepare the insert query.
+		$query
+			->insert($db->quoteName('#__share_draft'))
```

Comment:
This needs to be done by using a JTable class instead of a direct query to the DB ?
https://docs.joomla.org/Using_the_JTable_class

Note that the load() method in the JTable class can also be used to retrieve data 

cc @roland-d